async_tcp_server.cpp

```diff
@@ -0,0 +1,101 @@
+#include "async_tcp_server.hpp"
+#include <algorithm>
+#include <cstring>
+#include <iostream>
+#include <sys/select.h>
+
+AsyncTCPServer::AsyncTCPServer() {}
+
+AsyncTCPServer::AsyncTCPServer(Socket_sptr master_socket, fd_set fdsetRead): m_fdsetRead(fdsetRead)
+{
+    FD_ZERO(&m_fdsetRead);
+    FD_SET(master_socket->GetSocketHandle(), &m_fdsetRead);
```

Comment:
One more suggestion to think about: maybe we want to add conversion operator to `Socket` class to make it useful in contexts like this?